codexctl/device.py

```
@@ -416,6 +422,86 @@ def restore_previous_version(self) -> None:
+
+        print(f"Uploading {file_location} to {destination}")
+
+        ftp_client.put(
+            file_location, destination, callback=self.output_put_progress
+        )
+
+    def install_manual_update(self, version_file: str) -> None:
+        if self.client:
+            print(f"Uploading {version_file} image")
+
+            out_image_file = f"/tmp/{out_image_file}"
```

Ruff (0.8.0)

Undefined name `out_image_file`

(F821)